src/game_engine/Space_Game.java

```diff
@@ -251,15 +251,18 @@ private void deleteProjectiles() {
 		while(projIterator.hasNext()) {
 			
 			Projectile tempProj = projIterator.next();
-			if(tempProj.getyloc() < 10.0) {
+			if(tempProj.getyloc() < 10.0 || tempProj.getxloc() > 595) {
 				projIterator.remove();			
 			}
 		}
 		
-		
-		for(Projectile aProj: playerProjectiles) {
-			if(aProj.getyloc() < 10.0) {
-				
+		projIterator = enemyProjectiles.listIterator();
+
+		while(projIterator.hasNext()) {

```

Comment:
create a flag in the projectile function that gets raised when outof bounds, that way most of the stuff is handled in the projectile class rather than in the game class.